File: data/translations.py
"""
النصوص بالعربية والإنجليزية
"""

TRANSLATIONS = {
    'ar': {
        'app_title': 'شجرة أنساب العرب',
        'hero_title': 'رحلة عبر التاريخ',
        'hero_subtitle': 'من إسماعيل عليه السلام إلى محمد صلى الله عليه وسلم',
        'cta_button': 'ابدأ الاستكشاف',
        
        # Stats
        'stat_generations': 'جيل',
        'stat_personalities': 'شخصية',
        'stat_companions': 'صحابي',
        'stat_poets': 'شاعر',
        
        # About
        'about_title': 'نبذة عن المشروع',
        'about_text': '''
        مشروع شجرة أنساب العرب هو رحلة تفاعلية لاستكشاف أصول العرب
        من النبي إسماعيل عليه السلام وصولاً إلى النبي محمد صلى الله عليه وسلم.
        يهدف المشروع إلى حفظ التاريخ وتسهيل فهم الأنساب والعلاقات بين
        الشخصيات التاريخية المهمة.
        ''',
        
        # Header
        'toggle_theme': 'تبديل الوضع الليلي',
        'change_language': 'English',

        # Tree view - navigation
        'tree_title': 'شجرة الأنساب',
        'back_to_landing': 'الرئيسية',
        'load_next_generation': 'تحميل الجيل التالي',
        'no_more_generations': 'وصلت إلى آخر جيل',
        'generation_label': 'الجيل',

        # Tree view - sidebar
        'sidebar_title': 'الفلاتر والبحث',
        'search_placeholder': 'ابحث باسم الشخص...',
        'filter_all': 'الكل',
        'statistics': 'الإحصائيات',
        'favorites': 'المفضلة',
        'no_results': 'لا توجد نتائج',
        'no_favorites': 'لا توجد مفضلة',

        # Tree view - legend / types
        'legend': 'دليل الألوان',
        'type_prophet': 'نبي',
        'type_companion': 'صحابي',
        'type_poet': 'شاعر',
        'type_leader': 'قائد',
        'type_scholar': 'عالم',

        # Person modal
        'close': 'إغلاق',
        'biography': 'السيرة',
        'birth_year': 'سنة الميلاد',
        'death_year': 'سنة الوفاة',
        'children': 'الأبناء',
        'add_favorite': 'إضافة للمفضلة',
        'remove_favorite': 'إزالة من المفضلة',
        'unknown': 'غير معروف',
    },

    'en': {
        'app_title': 'Arab Genealogy Tree',
        'hero_title': 'A Journey Through History',
        'hero_subtitle': 'From Prophet Ismail (PBUH) to Prophet Muhammad (PBUH)',
        'cta_button': 'Start Exploring',
        
        # Stats
        'stat_generations': 'Generations',
        'stat_personalities': 'Personalities',
        'stat_companions': 'Companions',
        'stat_poets': 'Poets',
        
        # About
        'about_title': 'About the Project',
        'about_text': '''
        The Arab Genealogy Tree project is an interactive journey to explore
        the origins of Arabs from Prophet Ismail (PBUH) to Prophet Muhammad (PBUH).
        The project aims to preserve history and facilitate understanding of
        lineages and relationships between important historical figures.
        ''',
        
        # Header
        'toggle_theme': 'Toggle Dark Mode',
        'change_language': 'عربي',

        # Tree view - navigation
        'tree_title': 'Genealogy Tree',
        'back_to_landing': 'Home',
        'load_next_generation': 'Load Next Generation',
        'no_more_generations': 'You have reached the last generation',
        'generation_label': 'Generation',

        # Tree view - sidebar
        'sidebar_title': 'Filters & Search',
        'search_placeholder': 'Search by name...',
        'filter_all': 'All',
        'statistics': 'Statistics',
        'favorites': 'Favorites',
        'no_results': 'No results found',
        'no_favorites': 'No favorites yet',

        # Tree view - legend / types
        'legend': 'Legend',
        'type_prophet': 'Prophet',
        'type_companion': 'Companion',
        'type_poet': 'Poet',
        'type_leader': 'Leader',
        'type_scholar': 'Scholar',

        # Person modal
        'close': 'Close',
        'biography': 'Biography',
        'birth_year': 'Birth Year',
        'death_year': 'Death Year',
        'children': 'Children',
        'add_favorite': 'Add to Favorites',
        'remove_favorite': 'Remove from Favorites',
        'unknown': 'Unknown',
    }
}


def get_text(key: str, lang: str = 'ar') -> str:
    """
    الحصول على النص حسب اللغة
    """
    return TRANSLATIONS.get(lang, {}).get(key, key)

File: frontend/lib/translations.ts
import { Language } from './types';

const T: Record<string, Record<Language, string>> = {
  app_title:            { ar: 'شجرة النسب العربية',        en: 'Arab Genealogy Tree' },
  app_subtitle:         { ar: 'من إبراهيم الخليل إلى محمد ﷺ', en: 'From Ibrahim to Muhammad ﷺ' },
  view_tree:            { ar: 'استعرض الشجرة',             en: 'View the Tree' },
  stat_generations:     { ar: 'جيلاً',                     en: 'Generations' },
  stat_personalities:   { ar: 'شخصية',                    en: 'Personalities' },
  stat_prophets:        { ar: 'أنبياء',                    en: 'Prophets' },
  stat_companions:      { ar: 'صحابة وأمهات مؤمنين',      en: 'Companions & Wives' },
  about_title:          { ar: 'عن المشروع',                en: 'About This Project' },
  about_body:           { ar: 'تتتبع هذه الشجرة النسب العربي من إبراهيم الخليل عليه السلام مروراً بإسماعيل والعرب العدنانية حتى خاتم الأنبياء محمد ﷺ وصحابته الكرام.', en: 'This tree traces Arab lineage from Ibrahim (PBUH) through Ismail and the Adnanite Arabs to the Seal of the Prophets Muhammad ﷺ and his noble companions.' },
  // Tree page
  load_next:            { ar: 'تحميل الجيل التالي',        en: 'Load Next Generation' },
  generation:           { ar: 'الجيل',                     en: 'Generation' },
  no_more:              { ar: 'تم تحميل كامل الشجرة',      en: 'Full tree loaded' },
  // Legend
  legend_prophet:       { ar: 'نبي',                       en: 'Prophet' },
  legend_companion:     { ar: 'صحابي',                     en: 'Companion' },
  legend_leader:        { ar: 'قائد',                      en: 'Leader' },
  legend_poet:          { ar: 'شاعر',                      en: 'Poet' },
  legend_scholar:       { ar: 'عالم',                      en: 'Scholar' },
  // Ancestor comparison
  hint_select:          { ar: 'انقر على بطاقة لتتبع مسارها، أو اختر أكثر من شخص لإيجاد الجد المشترك', en: 'Click a card to trace its path, or select multiple cards to find their common ancestor' },
  selected_of:          { ar: 'محدد',                      en: 'selected' },
  clear_all:            { ar: 'مسح الكل',                  en: 'Clear All' },
  max_reached:          { ar: 'وصلت للحد الأقصى',          en: 'Maximum reached' },
  // Modal
  born:                 { ar: 'وُلد',                      en: 'Born' },
  died:                 { ar: 'تُوفّي',                    en: 'Died' },
  close:                { ar: 'إغلاق',                     en: 'Close' },
  generation_label:     { ar: 'الجيل',                     en: 'Generation' },
  add_favorite:         { ar: 'أضف للمفضلة',               en: 'Add to favourites' },
  remove_favorite:      { ar: 'إزالة من المفضلة',          en: 'Remove from favourites' },
  // Sidebar
  search_placeholder:   { ar: 'ابحث عن شخص...',            en: 'Search a person...' },
  filter_all:           { ar: 'الكل',                      en: 'All' },
  filter_prophets:      { ar: 'الأنبياء',                  en: 'Prophets' },
  filter_companions:    { ar: 'الصحابة',                   en: 'Companions' },
  filter_leaders:       { ar: 'القادة',                    en: 'Leaders' },
  no_results:           { ar: 'لا توجد نتائج',             en: 'No results' },
  language:             { ar: 'English',                   en: 'العربية' },
  dark_mode:            { ar: 'الوضع الداكن',              en: 'Dark mode' },
};

export function t(key: string, lang: Language): string {
  return T[key]?.[lang] ?? key;
}
